### PR TITLE
Applet Bar improvement

### PR DESCRIPTION
**Pull request by m-novikov**
This is an implementation of numbers and open/closed icons in applet bar.
The current implementation of AppletDrawerToolBox had some problems, so I was forced to split it into AppletBarMngr that is responsible for tracking applet and AppletBarToolBox that contains presentation logic.
What's left:
[ ] Write tests for AppletBarMngr
[ ] Write tests for AppletDrawerToolBox
[ ] Maybe change icons to something better

### CODE REVIEW

**Comment by akreshuk** [comment]
could you post a screenshot of what it looks like now?

**Comment by m-novikov** [comment]
ilastik_applet_bar_numbers_collapsible

**Comment by m-novikov** [comment]
It is possible to do it without icons, only using UTF-8 symbols. But the added benefit of icons is that it is the only way to make tab area bigger. Currently, it is quite narrow for a header.

**Comment by akreshuk** [comment]
Looks very nice!

File: ilastik/shell/gui/ilastikShell.py [hunk 1]
```diff
@@ -1064,9 +1066,8 @@ def handleAppletBarItemExpanded(self, modelIndex):
         """
         The user wants to view a different applet bar item.
         """
-        drawerIndex = modelIndex
-        if drawerIndex != -1:
-            self.setSelectedAppletDrawer(drawerIndex)
+        print("Switch to", modelIndex)
```
Comment (k-dominik) [inline ilastik/shell/gui/ilastikShell.py]: should be removed before merge

**Pull request merged**